test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ControllerTest.cs

```diff
@@ -274,6 +276,48 @@ public void TempData_CanSetAndGetValues()
             Assert.Equal(input, result);
         }
 
+        public static TheoryData<object, Type> IncompatibleModelData
+        {
+            get
+            {
+                // Small grab bag of instances and expected types with no common base except typeof(object).
+                return new TheoryData<object, Type>
+                {
+                    { null, typeof(object) },
+                    { true, typeof(bool) },
+                    { 43.78, typeof(double) },
+                    { "test string", typeof(string) },
+                    { new List<int>(), typeof(List<int>) },
+                    { new List<string>(), typeof(List<string>) },
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(IncompatibleModelData))]
+        public void ViewDataModelSetter_DoesNotThrow(object model, Type expectedType)
+        {
+            // Arrange
+            var activator = new ViewDataDictionaryControllerPropertyActivator(new EmptyModelMetadataProvider());
```

Comment:
Test 1: construct a controller and look at the metadata/type of the VDD
Test 2: construct a VDD matching what controller would have a set the model.

I think it's fine if you want to leave this here as an integration test, it just seems like it's covering a few different things.